packages/node/src/core/providers/state.ts

```diff
@@ -23,17 +24,19 @@ export function createEVMState(
     blockHistoryLimit: chainProvider.blockHistoryLimit || 600,
     chainId: chain.id,
     chainType: 'evm' as ChainType,
-    logFormat: settings.logFormat,
+    logFormat: config.nodeSettings.logFormat,
     minConfirmations: chainProvider.minConfirmations || 6,
     name: chainProvider.name,
-    providerId: settings.providerId,
+    // TODO: derive from mnemonic
```

Comment:
It would be good to derive `providerIdShort` with `providerId`. It feels like we may need to change the derivation rule (first 7 characters of `providerId`) in the future so doing the derivation once at the start will make it easier to modify.
I also need this `providerIdShort` to invoke the Lambda function.

Reply:
Or I guess `providerIdShort` would need to be in the coordinator state

Reply:
I don't think we need to keep `providerId` or `providerIdShort` in the state anymore. If we're deriving it from `MASTER_KEY_MNEMONIC`, then we can just derive it when needed. Something like:

```ts
export function getMasterWallet(provider?: ethers.providers.JsonRpcProvider) {
  const mnemonic = config.getMasterKeyMnemonic();
  const masterHdNode = ethers.utils.HDNode.fromMnemonic(mnemonic);
  return new ethers.Wallet(masterHdNode.privateKey, provider);
}

export function getProviderId(provider?: ethers.providers.JsonRpcProvider) {
  const masterWallet = getMasterWallet(provider);
  return ethers.utils.keccak256(ethers.utils.defaultAbiCoder.encode(['address'], [masterWallet.address]));
}

export function getProviderIdShort(provider?: ethers.providers.JsonRpcProvider) {
  return getProviderId(provider).substring(2, 9);
}
```
What do you think?

Reply:
We shouldn't need a provider to derive `providerId`, this also works
```js
const masterHdNode = ethers.utils.HDNode.fromMnemonic(mnemonic);
const providerId = ethers.utils.keccak256(ethers.utils.defaultAbiCoder.encode(['address'], [masterHdNode.address]));
```
but it still takes almost 80ms. Deriving from the private key directly
```js
const masterWallet = new ethers.Wallet(privateKey);
const providerId = ethers.utils.keccak256(ethers.utils.defaultAbiCoder.encode(['address'], [masterWallet.address]));
```
takes 55ms so it's not only about the mnemonic. It's the first line that uses 90%+ of that time, which makes me think that we should use a common `HDNode` for deriving different `designatedWallet`s (maybe something to think about for the future).

I guess we will run a profiler at some point to detect these inefficiencies but we can be preemptive about this one. `providerId` gets used a lot and there is one per node so no reason not to derive it only once (though deriving `providerIdShort` should be fine).

Reply:
Thanks. I'm removing the `provider` argument for now as it's unnecessary

Even 55ms is a long time to derive the `providerId` so it's probably better to cache this (and a common `HDNode`) in the state. I'll make this change now